foods/views.py

```diff
@@ -0,0 +1,90 @@
+from django.http import Http404
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from rest_framework import status
+from django.conf import settings
+from rest_framework.parsers import MultiPartParser, FormParser
+from django.core.paginator import PageNotAnInteger, EmptyPage, Paginator
+from .serializers import FoodSerializer, PostOrPutFoodSerializer, ImagesSerializer
+from .models.CategoryModel import Category
+from .models.FoodModel import Food
+from .models.ImagesMopdel import Images
+
+class FoodList(APIView):
+    serializer_class = FoodSerializer
+
+    def get(self, request, format=None):
+        foods = Food.objects.all().order_by('-id')
+        pagesize= int(settings.PAGESIZE)
+        pagetotal= round(Food.objects.all().count()/pagesize + 0.5)
+        paginator = Paginator(foods, pagesize)
+        page = request.GET.get("page")
+        try: 
+            paginated_querySet=paginator.page(page)
+        except PageNotAnInteger:
+            page=1
+            paginated_querySet=paginator.page(page)
+        except EmptyPage:
+            paginated_querySet=paginator.page(paginator.num_pages)
+        serializer = self.serializer_class(paginated_querySet, many=True)
+        content = {
+            "page": page,
+            "pagetotal": pagetotal,
+            "listfoods": serializer.data,
+        }
+        return Response(content)
+
+
+
+class CreateFood(APIView):
+    parser_classes = (MultiPartParser, FormParser)
+    serializer_class = PostOrPutFoodSerializer
+
+    def post(self, request, format=None):
+        files = request.FILES.getlist("files")
+        serializer = self.serializer_class(data=request.data)
+        if serializer.is_valid():
```

Comment:
Can you raise an exception in `is_valid`?

Reply:
is serializer.is_valid(raise_exception=True)